docs/tools/doc_generator.py

```diff
@@ -0,0 +1,106 @@
+import ast
```

Comment:
1. Check license of the source sample files. Make sure it's respected.
2. If we can copy the file, then add header on each python file.

Reply:
1. It's in the Python Standard Library and has the Python License.
2. Why do we need to copy the files?

Reply:
I mean the `doc_generator.py` itself, where is it reference? If it's written by you, you can just add header. But if it's referenced from other project, we need to check the license.

Reply:
Okay, it's written by me. I can add headers afterwards.